### PR TITLE
[17.0][MIG] sale_commission: Migration to 17.0

### PR DESCRIPTION
**Pull request by ambientWave**
I upgraded the 'sale_commission' module to be installed on Odoo 17.0. More focus was given to upgrading xml files.

The demo files weren't upgraded and hence, an error during installation is warning user that demo data wasn't populated. However, the module can be used without demo data.

Python code needs to be tested thoroughly.

All contributions and comments are welcomed.

Depends on

- https://github.com/OCA/commission/pull/497
- https://github.com/OCA/commission/pull/499

### CODE REVIEW

**Comment by ambientWave** [comment]
The following error from Odoo and OCB tests
```
odoo-addon-sale_commission @ file:///__w/commission/commission/sale_commission
+ pip install -r test-requirements.txt -c test-constraints.txt
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Looking in indexes: https://wheelhouse.odoo-community.org/oca-simple-and-pypi
ERROR: Could not find a version that satisfies the requirement odoo-addon-account_commission<17.1dev,>=17.0dev (from versions: 15.0.2.0.0, 15.0.2.0.1, 15.0.2.1.0, 15.0.2.1.0.2, 15.0.2.1.0.3, 15.0.2.1.0.5, 15.0.2.2.0, 15.0.2.3.0, 15.0.2.3.1, 15.0.2.3.2, 15.0.2.3.3, 15.0.2.3.3.1, 15.0.2.3.3.2, 15.0.2.3.3.4, 15.0.3.0.0, 15.0.3.0.1, 15.0.3.1.0, 15.0.3.2.0, 16.0.1.0.0.2, 16.0.1.0.0.3, 16.0.1.0.0.4, 16.0.1.1.0, 16.0.1.1.1, 16.0.1.1.1.2, 16.0.1.1.1.4, 16.0.1.2.0, 16.0.1.2.0.1, 16.0.1.3.0, 16.0.1.4.0, 16.0.1.4.1, 16.0.1.4.1.1, 16.0.1.4.1.3, 16.0.1.4.1.4, 16.0.1.4.2, 16.0.1.4.2.3, 16.0.1.4.2.5, 16.0.2.0.0, 16.0.2.0.1, 16.0.2.1.0, 16.0.2.2.0)
ERROR: No matching distribution found for odoo-addon-account_commission<17.1dev,>=17.0dev
Error: Process completed with exit code 1.
```
@pedrobaeza @sbidoul Seems to be a dependency issue. How one can resolve this?

**Comment by sbidoul** [comment]
That is because the commission module is not merged yet.
See https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

**Comment by ambientWave** [comment]
@pedrobaeza Currently, there's a method named 'action_switch_invoice_into_refund_credit_note' that is called from 'account.move' model which seems to exist no more.
What you would propose for a resolution?
I ported the method from odoo 16.0 and pasted it inside the inherited account.move

**Comment by Tisho99** [comment]
Hi @ambientWave 

Can you update this PR and fix the commit history?

There should be only 3 commits:

[IMP] sale_commission: pre-commit auto fixes
[MIG] sale_commission: Migration to 17.0
[DON'T MERGE] test-requirements.txt

Thank you!

File: sale_commission/views/sale_order_view.xml
Comment (Tisho99) [inline sale_commission/views/sale_order_view.xml]:
```suggestion
                    invisible="commission_free or invoice_lines"
```

File: sale_commission/views/sale_order_view.xml
Comment (Tisho99) [inline sale_commission/views/sale_order_view.xml]:
```suggestion
                <field name="invoice_lines" column_invisible="1" />
```

File: sale_commission/views/sale_order_view.xml
Comment (Tisho99) [inline sale_commission/views/sale_order_view.xml]:
```suggestion
                <field name="commission_free" column_invisible="1" />
```

**Review by Tisho99 — CHANGES_REQUESTED** [review]

**Comment by ambientWave** [comment]
@Tisho99 

Commit history has been cleaned up!

File: sale_commission/views/sale_order_view.xml
Comment (ambientWave) [reply sale_commission/views/sale_order_view.xml]: @Tisho99 

All changes were applied

File: sale_commission/views/sale_order_view.xml
Comment (Tisho99) [inline sale_commission/views/sale_order_view.xml]:
```suggestion
                    invisible="commission_free or invoice_lines"
```

Comment (Tisho99) [reply sale_commission/views/sale_order_view.xml]: Why haven't you applied this change?

**Review by Tisho99 — CHANGES_REQUESTED** [review]

**Review by Tisho99 — APPROVED** [review]
LGTM

**Review by luis-ron — APPROVED** [review]
LGTM 👍🏻

**Comment by Tisho99** [comment]
Hi @ambientWave 

account_commission has been merged,

can you remove the dependencies commit to merge this PR?

**Comment by ambientWave** [comment]
@pedrobaeza @sbidoul @Tisho99 @luis-ron @jimhuds @Jaimermaccione

I want to thank you for your efforts and allowing people around the globe to contribute.

It was an honor.

**Comment by pedrobaeza** [comment]
Thanks to you for doing the contribution. I hope it won't be the last!

/ocabot migration sale_commission
/ocabot merge nobump

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
